vault/core_cache_invalidate.go

```diff
@@ -68,5 +68,7 @@ func (c *Core) invalidateInternal(ctx context.Context, key string) error {
 		c.logger.Warn("no idea how to invalidate cache. Maybe it's not cached and this is fine, maybe not", "key", key)
 	}
 
+	c.router.Invalidate(ctx, key)
```

Comment:
Won't we get the warning above here? Perhaps we should return a boolean as to whether we found a matching entry and add it to the switch above?